pkg/prometheus/promcfg.go

```diff
@@ -1248,7 +1351,14 @@ func (cg *ConfigGenerator) generateServiceMonitorConfig(
 	assetKey := fmt.Sprintf("serviceMonitor/%s/%s/%d", m.Namespace, m.Name, i)
 	cfg = cg.addOAuth2ToYaml(cfg, ep.OAuth2, store.OAuth2Assets, assetKey)
 
-	cfg = addTLStoYaml(cfg, m.Namespace, ep.TLSConfig)
+	if ep.TLSConfig != nil {
+		cfg = addTLStoYaml(cfg, m.Namespace, ep.TLSConfig)
+	}
+
+	if scrapeClass != nil {
+		cfg = addScrapeClassTLSConfigToYaml(cfg, m.Namespace, scrapeClass)
```

Comment:
IMHO it would be better/simpler to merge `ep.TLSConfig` with the selected scrape class before and then call `addTLStoYaml` on the result.

Reply:
I was working on that.
But we have differences between service monitor and other monitors.
Service monitor allows a TLSConfig while the other allow only a SafeTLS, and using this approach will lead us with two different code one to merge TLS and other to merge SafeTLS config.

I was thinking about keep this implementation since it's more generic and work as expected, wdyt?